identifier/identifier.go

```diff
@@ -75,3 +80,70 @@ func NewIP(ip netip.Addr) ACMEIdentifier {
 		Value: ip.String(),
 	}
 }
+
+// FromCert extracts the Subject Alternative Names from a certificate, and
+// returns a slice of ACMEIdentifiers and an error. It does not extract the
+// Subject Common Name.
```

Comment:
It seems like it _could_ inspect the CN -- passing the result through Normalize will remove any duplicate introduced by doing so. What's the reasoning for not doing that here?

Reply:
That arose from [other review feedback](https://github.com/letsencrypt/boulder/pull/7961#discussion_r1955179676) but the robustness doesn't cost much, and adding it back in will simplify the tests' logic if we test `FromCert` too.